packages/@n8n/task-runner/src/task-runner.ts

```diff
@@ -108,6 +113,17 @@ export abstract class TaskRunner {
 		});
 		this.ws.addEventListener('message', this.receiveMessage);
 		this.ws.addEventListener('close', this.stopTaskOffers);
+		this.resetIdleTimer();
+	}
+
+	private resetIdleTimer() {
+		if (this.idleTimeout === 0) return;
+
+		this.clearIdleTimer();
+
+		this.idleTimer = setTimeout(() => {
+			if (this.runningTasks.size === 0) process.exit(0);
```

Comment:
This works, but since we have other resources as well we should do a graceful shutdown. Meaning gracefully shutdown:
1. sentry, so all pending error reports are sent
2. healthcheck server

We could make `TaskRunner` an `EventEmitter` and emit an `exit` event, or provide a callback function in the constructor. Then in the `start.ts` we can do a graceful shutdown.